wardrobe/tradable_kitties/src/lib.rs

```diff
@@ -0,0 +1,573 @@
+            ConstraintCheckerError::TwoMomsNotValid => {
+                TradableKittyConstraintCheckerError::TwoMomsNotValid
+            }
+            ConstraintCheckerError::TwoDadsNotValid => {
+                TradableKittyConstraintCheckerError::TwoDadsNotValid
+            }
+            ConstraintCheckerError::NewMomIsStillRearinToGo => {
+                TradableKittyConstraintCheckerError::NewMomIsStillRearinToGo
+            }
+            ConstraintCheckerError::NewDadIsStillRearinToGo => {
+                TradableKittyConstraintCheckerError::NewDadIsStillRearinToGo
+            }
+            ConstraintCheckerError::NewParentFreeBreedingsIncorrect => {
+                TradableKittyConstraintCheckerError::NewParentFreeBreedingsIncorrect
+            }
+            ConstraintCheckerError::NewParentDnaDoesntMatchOld => {
+                TradableKittyConstraintCheckerError::NewParentDnaDoesntMatchOld
+            }
+            ConstraintCheckerError::NewParentNumberBreedingsIncorrect => {
+                TradableKittyConstraintCheckerError::NewParentNumberBreedingsIncorrect
+            }
+            ConstraintCheckerError::NewChildDnaIncorrect => {
+                TradableKittyConstraintCheckerError::NewChildDnaIncorrect
+            }
+            ConstraintCheckerError::NewChildFreeBreedingsIncorrect => {
+                TradableKittyConstraintCheckerError::NewChildFreeBreedingsIncorrect
+            }
+            ConstraintCheckerError::NewChildHasNonZeroBreedings => {
+                TradableKittyConstraintCheckerError::NewChildHasNonZeroBreedings
+            }
+            ConstraintCheckerError::NewChildIncorrectParentInfo => {
+                TradableKittyConstraintCheckerError::NewChildIncorrectParentInfo
+            }
+            ConstraintCheckerError::TooManyBreedingsForKitty => {
+                TradableKittyConstraintCheckerError::TooManyBreedingsForKitty
+            }
+            ConstraintCheckerError::NotEnoughFreeBreedings => {
+                TradableKittyConstraintCheckerError::NotEnoughFreeBreedings
+            }
+            ConstraintCheckerError::MintingNothing => {
+                TradableKittyConstraintCheckerError::MintingNothing
+            }
+            ConstraintCheckerError::MintingWithInputs => {
+                TradableKittyConstraintCheckerError::MintingWithInputs
+            }
+        }
+    }
+}
+
+#[derive(
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Clone,
+    Encode,
+    Decode,
+    Hash,
+    Debug,
+    TypeInfo,
+)]
+pub enum TradableKittyConstraintChecker<const ID: u8> {
+    /// A mint transaction that creates Tradable Kitties
+    Mint,
+    /// A typical Breed transaction where kitties are consumed and new family(Parents(mom,dad) and child) is created.
+    Breed,
+    ///Update various properties of kitty.
+    UpdateProperties,
+    ///Can buy a new kitty from others
+    Buy,
+}
```

Comment:
As I mentioned above, I think you should change your variants here. Rather than minting, you should have a variant that allows you to list an existing kitty for sale.

Also you can remove Breed (or make it a simple wrapper) because it already exists in the existing kitty code.

Reply:
Yes, 
Will remove Breed and Mint .
Will Add ListForSale and DelistFromSale.